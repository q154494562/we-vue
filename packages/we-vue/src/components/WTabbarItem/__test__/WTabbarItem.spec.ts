import { shallowMount } from '@vue/test-utils'
import TabbarItem from '../'

describe('tabbar-item', () => {
  test('create', () => {
    const wrapper = shallowMount(TabbarItem, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-tabbar-item')
    expect(wrapper.classes()).toContain('weui-tabbar__item')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('text', () => {
    const wrapper = shallowMount(TabbarItem, {
      slots: {
        default: 'test',
      },
    })

    expect(wrapper.find('p.weui-tabbar__label').text()).toBe('test')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('isOn', () => {
    const wrapper = shallowMount(TabbarItem, {
      propsData: {
        isOn: true,
      },
    })

    expect(wrapper.classes()).toContain('weui-bar__item_on')
  })

  test('handle click', () => {
    const routeLinkSpy = jest.fn()
    const wrapper = shallowMount(TabbarItem, {
      propsData: {},
      methods: {
        routeLink: routeLinkSpy,
      },
    })

    wrapper.trigger('click')
    expect(wrapper.emitted().click).toBeTruthy()
    expect(routeLinkSpy).toHaveBeenCalled()
  })
})
