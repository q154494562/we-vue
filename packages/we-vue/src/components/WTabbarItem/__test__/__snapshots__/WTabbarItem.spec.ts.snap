// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tabbar-item create 1`] = `
<div class="weui-tabbar__item">
  <p class="weui-tabbar__label"></p>
</div>
`;

exports[`tabbar-item text 1`] = `
<div class="weui-tabbar__item">
  <p class="weui-tabbar__label">test</p>
</div>
`;
